Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -646,558 +587,373 @@ protected virtual void Dispose(bool disposing)
+            result = IsIPv4Enabled && !IsIPv6Enabled ? "127.0.0.1" : "::1";
+            _logger.LogWarning("{Source}: Only loopback {Result} returned, using that as bind address.", source, result);
+            return result;
         }
 
-        /// <summary>
-        /// Initialises the remote address values.
-        /// </summary>
-        private void InitialiseRemote(NetworkConfiguration config)
+        /// <inheritdoc/>
+        public IReadOnlyList<IPData> GetInternalBindAddresses()
         {
-            lock (_intLock)
-            {
-                RemoteAddressFilter = CreateIPCollection(config.RemoteIPFilter);
-            }
+            // Select all local bind addresses
+            return _interfaces.Where(x => IsInLocalNetwork(x.Address))
+                .OrderBy(x => x.Index)
+                .ToList();
         }
 
-        /// <summary>
-        /// Initialises internal LAN cache settings.
-        /// </summary>
-        private void InitialiseLAN(NetworkConfiguration config)
+        /// <inheritdoc/>
+        public bool IsInLocalNetwork(string address)
         {
-            lock (_intLock)
+            if (NetworkExtensions.TryParseToSubnet(address, out var subnet))
             {
-                _logger.LogDebug("Refreshing LAN information.");
-
-                // Get configuration options.
-                string[] subnets = config.LocalNetworkSubnets;
-
-                // Create lists from user settings.
-
-                _lanSubnets = CreateIPCollection(subnets);
-                _excludedSubnets = CreateIPCollection(subnets, true).AsNetworks();
-
-                // If no LAN addresses are specified - all private subnets are deemed to be the LAN
-                _usingPrivateAddresses = _lanSubnets.Count == 0;
+                return IPAddress.IsLoopback(subnet.Prefix) || (_lanSubnets.Any(x => x.Contains(subnet.Prefix)) && !_excludedSubnets.Any(x => x.Contains(subnet.Prefix)));
+            }
 
-                // NOTE: The order of the commands generating the collection in this statement matters.
-                // Altering the order will cause the collections to be created incorrectly.
-                if (_usingPrivateAddresses)
+            if (NetworkExtensions.TryParseHost(address, out var addresses, IsIPv4Enabled, IsIPv6Enabled))
+            {
+                bool match = false;
+                foreach (var ept in addresses)
                 {
-                    _logger.LogDebug("Using LAN interface addresses as user provided no LAN details.");
-                    // Internal interfaces must be private and not excluded.
-                    _internalInterfaces = CreateCollection(_interfaceAddresses.Where(i => IsPrivateAddressRange(i) && !_excludedSubnets.ContainsAddress(i)));
+                    match = match || IPAddress.IsLoopback(ept) || (_lanSubnets.Any(x => x.Contains(ept)) && !_excludedSubnets.Any(x => x.Contains(ept)));
```

Comment:
No need to store this in a variable outside the loop. Once it is true, you break and return true. Just put the expression in the `if` and return true once it's true. Return false at the end.

Reply:
You could even simplify the code by using [`List.Exists`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.exists?view=net-7.0) or [`Array.Exists`](https://learn.microsoft.com/en-us/dotnet/api/system.array.exists?view=net-7.0) depending on the type. E.g.

```C#
return Array.Exists(addresses, ept => IPAddress.IsLoopback(ept) || (_lanSubnets.Any(x => x.Contains(ept)) && !_excludedSubnets.Any(x => x.Contains(ept)));